core/livepeernode.go

```diff
@@ -232,9 +232,18 @@ func (n *LivepeerNode) TranscodeAndBroadcast(config net.TranscodeConfig, cm eth.
 			glog.Errorf("Error decoding byte array into segment: %v", err)
 		}
 		glog.V(common.DEBUG).Infof("Decoding of segment took %v", time.Since(start))
-		n.transcodeAndBroadcastSeg(&ss.Seg, ss.Sig, cm, t, resultStrmIDs, broadcasters, config)
-		glog.V(common.DEBUG).Infof("Encoding and broadcasting of segment %v took %v", ss.Seg.SeqNo, time.Since(start))
-		glog.V(common.SHORT).Infof("Finished transcoding segment %v, overall took %v\n\n\n", seqNo, time.Since(totalStart))
+
+		//If running in on-chain mode, check that segment was signed by broadcaster ETH address
+		segHash := (&ethTypes.Segment{StreamID: config.StrmID, SegmentSequenceNumber: big.NewInt(int64(seqNo)), DataHash: crypto.Keccak256Hash(ss.Seg.Data)}).Hash()
+		if cm == nil || (cm.BroadcasterAddr() == common.Address{}) || eth.VerifySig(cm.BroadcasterAddr(), segHash.Bytes(), ss.Sig) {
```

Comment:
The additional check `cm.BroadcasterAddr() == common.Address{}` fixes the CI tests since the `StubClaimManager` will return `common.Address{}` for `cm.BroadcasterAddr()`. In practice, I don't think it should ever be possible for `cm.BroadcasterAddr()` to return `common.Address{}` since the null address cannot create a job...

Reply:
Unrelated to this actual PR, but while reviewing/studying this bit of code, it might make things easier to reason about if `cm.AddReceipt` was hoisted up into `TranscodeAndBroadcast` ;  this would remove the need to pass in `cm` and `config` into transcodeAndBroadcastSeg (and another nil check in there). Basically, keeps this bit of `cm`-checking logic self contained, rather than propagating assumptions about `cm`'s state downstream.

Reply:
Yeah I agree that the checking logic for `cm` could be cleaned up a bit...though I think at the moment `cm.AddReceipt` has to be in `transcodeAndBroadcastSeg` because it requires the transcoded data to be passed in as a parameter